File: DSA Practice/LeetCode/1456. Maximum Number of Vowels in a Substring of Given Length/solution.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        
        maximum = 0

        #Using sliding window
        
        left = 0
        right = 0
        vc = 0

        while (right < len(s)):
            # check if the rightmost character is a vowel
            if s[right] in "aeiou":
                vc += 1
            
            # check if the window size is greater than k, 
            # and update vc and left pointer accordingly
            if right - left + 1 > k:
                if s[left] in "aeiou":
                    vc -= 1
                left += 1
                
            right += 1
            
            maximum = max(vc, maximum)

        return maximum

File: DSA Practice/LeetCode/1456. Maximum Number of Vowels in a Substring of Given Length/solution.java
class Solution {
    public int maxVowels(String s, int k) {
        int left = 0;
        int right = 0;
        int vc = 0;
        int maximum = 0;

        while (right < s.length()){

            char cr = s.charAt(right);
            char cl = s.charAt(left);

            // Right check
            if (cr == 'a' | cr == 'e' | cr == 'i' | cr == 'o' | cr == 'u'){
                vc++;
            }

            // Moving left if window crossed

            if (right - left + 1 > k){
                if (cl == 'a' | cl == 'e' | cl == 'i' | cl == 'o' | cl == 'u'){
                    vc--;
                }

                left++;
            }

            // Right increase
            right++;

            // Answer setting
            maximum = Math.max(vc, maximum);
            
        }

        return maximum;
    }
}